src/main/java/seedu/address/ui/TaskCardDetails.java

```diff
@@ -0,0 +1,196 @@
+        this.task = task;
+        // compulsory fields that are static in the fxml
+        setId(displayedIndex);
+        setTitle(task);
+        setDeadline(task);
+        setStartTime(task);
+        setStatus(task);
+        // Optional fields that are dynamically added. Order of methods (except tags) determine position.
+        setTagsIfPresent(task);
+        setDescriptionIfPresent(task);
+        setEmailIfPresent(task);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        // short circuit if same object
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof TaskCardDetails)) {
+            return false;
+        }
+
+        // state check
+        TaskCardDetails card = (TaskCardDetails) other;
+        return id.getText().equals(card.id.getText())
+                && task.equals(card.task);
+    }
+
+    private void setId(int displayedIndex) {
+        id.getStyleClass().add("cell_big_label");
+        id.setText(displayedIndex + ". ");
+    }
+
+    /**
+     * Sets text of the label to be title.
+     * Asserts that it is not blank because title cannot be created blank.
+     *
+     * @param task Task to be displayed.
+     */
+    private void setTitle(Task task) {
+        String titleValue = task.getTitle().fullTitle;
+        assert !titleValue.isBlank() : "Title cannot be displayed blank.";
+        title.getStyleClass().add("cell_big_label");
+        title.setText(titleValue);
+    }
+
+    /**
+     * Sets text of the label to be deadline.
+     * Asserts that it is not blank because deadline cannot be created blank.
+     *
+     * @param task Task to be displayed.
+     */
+    private void setDeadline(Task task) {
+        String deadlineValue = task.getDeadline().value;
+        assert !deadlineValue.isBlank() : "Deadline cannot be displayed blank.";
+        deadline.getStyleClass().add("cell_small_label");
+        deadline.setText(deadlineValue);
+    }
+
+    /**
+     * Sets text of the label to be startTime.
+     * Asserts that it is not blank because startTime cannot be created blank.
+     *
+     * @param task Task to be displayed.
+     */
+    private void setStartTime(Task task) {
+        String startTimeValue = task.getStartTime().value;
+        assert !startTimeValue.isBlank() : "StartTime cannot be displayed blank.";
+        starttime.getStyleClass().add("cell_small_label");
+        starttime.setText(startTimeValue);
+    }
+
+    /**
+     * Sets text of the label to be status.
+     * Asserts that it is not blank because status cannot be created blank.
+     * Sets style class according to its value.
+     *
+     * @param task Task to be displayed.
+     */
+    private void setStatus(Task task) {
+        String statusValue = task.getStatus().value;
+        assert !statusValue.isBlank() : "Status cannot be displayed blank.";
+
+        boolean isStatusDone = task.getStatus().isDone();
```

Comment:
Can consider refactor through law of demeter?

Reply:
Alright thanks!